text/3724-theupdateframework.md

```diff
@@ -0,0 +1,341 @@
+- Hardware keys shall be distributed for management and storage of signing keys by active members of the quorum. The Rust Foundation shall provide the hardware tokens at zero cost to the quorum members.
+- Hardware key loss or theft or similar will be handled by the quorum rotating out that key and rotating in a new key. Quorum members are responsible for reporting key loss or theft or similar to the infrastructure team in a timely fashion.
+
+# Reference Level Explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Repository Workflows
+[tuf-on-ci](https://github.com/theupdateframework/tuf-on-ci) shall be used for workflows on each repository.
+
+tuf-on-ci is a set of CI tools which are integrated with GitHub Actions for providing TUF quorum and hardware key support for managing TUF repositories via pull requests on GitHub. This is a ready made and production ready suite of CI that is used by the sigstore root signing for managing TUF quorums.
+
+### (tuf-root) Rustup Release
+
+- A signing event will be triggered via pull request in the `tuf-root` repository, which is assigned to the `Rustup Role` quorum, who must sign the request.
+
+### (tuf-root) Stable/Beta Release
+
+- A signing event will be triggered via pull request in the `tuf-root` repository, which is assigned to the `Release Role` quorum, who must sign the request.
+
+### (tuf-root) Nightly Release
+
+- An online signing delegate key will be utilized for signing updates to the repository. This key shall live in AWS KMS and allow for automated signing within the `tuf-root` repository.
+
+### (tuf-crates) Crate Release/Yank
+
+- An online signing delegate key will be utilized for signing updates to the repository. This key shall live in AWS KMS and allow for automated signing within the `tuf-crates` repository.
+
+### (tuf-root + tuf-crates) Crates.io Membership Change
+
+- The crates.io team will update the root role in the `tuf-crates` repository, triggering a signing event that the existing crates.io team must sign via Pull Request.
+- An update to the tuf-crates-root.json file will occur in the `tuf-root` repository, which shall trigger a new singing event Pull Request, which the root quorum must perform.
+
+## (cargo-tuf-lib) Standard TUF Implementation
+
+We propose creating a new crate, `cargo-tuf-lib`, which shall be used by both Cargo and Rustup for doing TUF synchronization and update procedures. This library shall be a shim wrapper around the `rust-tuf` crate (https://github.com/rustfoundation/rust-tuf), providing a simplified and shared interface for doing synchronization and verification of the TUF repositories and their files.
```

Comment:
How much influence/ownership do we have over `rust-tuf`? Cargo ends up needing to be rather opinionated about... a lot of things. What are dependencies are, and how they can be configured, and what their licenses are, etc.

Reply:
Is https://github.com/rustfoundation/rust-tuf supposed to be a fork of https://github.com/theupdateframework/rust-tuf which hasn't been created yet? It 404s.

cc @erickt

Reply:
It is a fork. It is just private right now, I believe, until this gets approved and implementation going. Nothing has been done on the fork just yet. (cc: @walterhpearce)

Reply:
We'd be happy to accept merges if you have anything to upload. I haven't been super active on rust-tuf since we haven't needed any changes for Fuchsia in a while. Also, please check out amazon's library https://crates.io/crates/tough as well, it also is pretty far along too.